Classes/Service/WebAuthnAuthenticationService.php

```diff
@@ -40,44 +43,92 @@ class WebAuthnAuthenticationService extends AuthenticationService
      */
     private $webAuthnSession;
 
-    public function __construct()
+    /**
+     * @var array
+     */
+    private $backendExtensionConfiguration;
+
+    /**
+     * @var PublicKeyCredentialSourceRepository
+     */
+    private $publicKeyCredentialSourceRepository;
+
+    public function __construct(
+        ?BackendUserRepository $backendUserRepository = null,
+        ?WebAuthnService $webAuthnService = null,
+        ?WebAuthnSession $webAuthnSession = null,
+        array $backendExtensionConfiguration = [],
+        ?PublicKeyCredentialSourceRepository $publicKeyCredentialSourceRepository = null
+    ) {
+        $this->backendUserRepository = $backendUserRepository ?? GeneralUtility::makeInstance(ObjectManager::class)->get(BackendUserRepository::class);
+        $this->webAuthnService = $webAuthnService ?? WebAuthnServiceFactory::fromGlobals();
+        $this->webAuthnSession = $webAuthnSession ?? GeneralUtility::makeInstance(WebAuthnSession::class);
+        $this->backendExtensionConfiguration = $backendExtensionConfiguration ?? GeneralUtility::makeInstance(ExtensionConfiguration::class)->get('cvc_webauthn');
+        $this->publicKeyCredentialSourceRepository = $publicKeyCredentialSourceRepository ?? GeneralUtility::makeInstance(ObjectManager::class)->get(PublicKeyCredentialSourceRepository::class);
+    }
+
+    public function processLoginData(array &$loginData, $passwordTransmissionStrategy)
     {
-        $this->backendUserRepository = GeneralUtility::makeInstance(ObjectManager::class)
-            ->get(BackendUserRepository::class);
-        $this->webAuthnService = WebAuthnServiceFactory::fromGlobals();
-        $this->webAuthnSession = GeneralUtility::makeInstance(WebAuthnSession::class);
+        $isProcessed = false;
+        if ($passwordTransmissionStrategy === 'normal') {
+            $loginData['webauthn-uident'] = GeneralUtility::_GP('webauthn-userident');
+            if ($loginData['uident'] == '') {
+                $loginData['uident'] = $loginData['webauthn-uident'];
+            } else {
+                $loginData['uident-text'] = $loginData['uident'];
```

Comment:
In TYPO3 the array key is named `uident_text` (with an `_` underscore instaed of a `-` minus).

Reply:
Oversight from me. Using underscores now. Thanks